xarray/backends/api.py

```diff
@@ -606,7 +606,7 @@ def open_mfdataset(paths, chunks=None, concat_dim=_CONCAT_DIM_DEFAULT,
     # Coerce 1D input into ND to maintain backwards-compatible API until API
     # for N-D combine decided
     # (see https://github.com/pydata/xarray/pull/2553/#issuecomment-445892746)
-    if concat_dim is None or concat_dim == _CONCAT_DIM_DEFAULT:
+    if concat_dim is None or concat_dim is _CONCAT_DIM_DEFAULT:
```

Comment:
Are we sure we want this to be an `is` test? Are we sure that the keyword argument will default to that string object?

Reply:
Yes, we use that as the default value here.

Possibly a better choice would be to use something like `ReprObject('<inferred>')`  for the sentinel object, e.g., as done here:
https://github.com/pydata/xarray/blob/69086b332c6c950587830b266df4e624c2106d89/xarray/core/dtypes.py#L8

Reply:
I guess what I am getting at is a matter of user-friendliness. A user looking up the call signature of this method will see the default as the string (and not an object), but, they'll never be able to explicitly force concat dimension inference, because passing the string themselves will not work (it'll be a different object).